File: app/src/main/java/ai/elimu/content_provider/room/dao/StoryBookDao.java
package ai.elimu.content_provider.room.dao;

import android.database.Cursor;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ai.elimu.content_provider.room.entity.StoryBook;

@Dao
public interface StoryBookDao {

    @Insert
    void insert(StoryBook storyBook);

    @Query("SELECT * FROM StoryBook WHERE id = :id")
    StoryBook load(Long id);

    @Query("SELECT * FROM StoryBook WHERE id = :id")
    Cursor loadAsCursor(Long id);

    @Query("SELECT * FROM StoryBook ORDER BY readingLevel,title")
    List<StoryBook> loadAll();

    @Query("SELECT * FROM StoryBook ORDER BY readingLevel,title")
    Cursor loadAllAsCursor();

    @Update
    void update(StoryBook storyBook);

    @Query("DELETE FROM StoryBook")
    void deleteAll();
}

File: app/src/main/java/ai/elimu/content_provider/room/dao/StoryBookDao.kt
package ai.elimu.content_provider.room.dao

import ai.elimu.content_provider.room.entity.StoryBook
import android.database.Cursor
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface StoryBookDao {
    @Insert
    fun insert(storyBook: StoryBook)

    @Query("SELECT * FROM StoryBook WHERE id = :id")
    fun load(id: Long?): StoryBook?

    @Query("SELECT * FROM StoryBook WHERE id = :id")
    fun loadAsCursor(id: Long?): Cursor

    @Query("SELECT * FROM StoryBook ORDER BY readingLevel,title")
    fun loadAll(): MutableList<StoryBook>

    @Query("SELECT * FROM StoryBook ORDER BY readingLevel,title")
    fun loadAllAsCursor(): Cursor

    @Update
    fun update(storyBook: StoryBook)

    @Query("DELETE FROM StoryBook")
    fun deleteAll()
}
